[CI] MixedClusterClientYamlTestSuiteIT test {p0=logsdb/10_settings/missing hostname field} failing

**Build Scans:**
- [elasticsearch-intake #16047 / 8.16.3_bwc-snapshots](https://gradle-enterprise.elastic.co/s/wd45sdz5wpxi6)
- [elasticsearch-intake #15984 / 8.17.1_bwc-snapshots](https://gradle-enterprise.elastic.co/s/sph4qyfmvr3c2)

**Reproduction Line:**
```
./gradlew ":qa:mixed-cluster:v8.16.3#mixedClusterTest" -Dtests.class="org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT" -Dtests.method="test {p0=logsdb/10_settings/missing hostname field}" -Dtests.seed=E55887F17FD44AD6 -Dtests.bwc=true -Dtests.locale=tzm-MA -Dtests.timezone=America/Matamoros -Druntime.java=23
```

**Applicable branches:**
8.x

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(test%20%7Bp0%3Dlogsdb%2F10_settings%2Fmissing%20hostname%20field%7D),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Failure at [logsdb/10_settings:98]: expected [2xx] status code but api [indices.create] returned [400 Bad Request] [{"error":{"root_cause":[{"type":"illegal_argument_exception","reason":"unknown index sort field:[host.name]","stack_trace":"org.elasticsearch.ElasticsearchException$1: unknown index sort field:[host.name]\n\tat org.elasticsearch.server@8.18.0-SNAPSHOT/org.elasticsearch.ElasticsearchException.guessRootCauses(ElasticsearchException.java:706)\n\tat org.elasticsearch.server@8.18.0-SNAPSHOT/org.elasticsearch.ElasticsearchException.generateFailureXContent(ElasticsearchException.java:634)\n\tat org.elasticsearch.server@8.18.0-SNAPSHOT/org.elasticsearch.rest.RestResponse.build(RestResponse.java:201)\n\tat org.elasticsearch.server@8.18.0-SNAPSHOT/org.elasticsearch.rest.RestResponse.<init>(RestResponse.java:160)\n\tat org.elasticsearch.server@8.18.0-SNAPSHOT/org.elasticsearch.rest.RestResponse.<init>(RestResponse.java:121)\n\tat org.elasticsearch.server@8.
[truncated]
```

**Issue Reasons:**
- [8.x] 2 failures in test test {p0=logsdb/10_settings/missing hostname field} (2.9% fail rate in 68 executions)
- [8.x] 2 failures in pipeline elasticsearch-intake (11.1% fail rate in 18 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -238,9 +238,6 @@ tests:
   - class: org.elasticsearch.search.basic.SearchWithRandomIOExceptionsIT
     method: testRandomDirectoryIOExceptions
     issue: https://github.com/elastic/elasticsearch/issues/118733
-  - class: org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT
-    method: test {p0=logsdb/10_settings/missing hostname field}
-    issue: https://github.com/elastic/elasticsearch/issues/120476
   - class: org.elasticsearch.action.search.SearchQueryThenFetchAsyncActionTests
     method: testBottomFieldSort
     issue: https://github.com/elastic/elasticsearch/issues/118214

